src/reanimated2/Hooks.js

```diff
@@ -582,6 +582,12 @@ export function useAnimatedReaction(prepare, react) {
   }
   const { inputs } = inputsRef.current;
 
+  if (dependencies === undefined) {
+    dependencies = [...inputs, prepare.__workletHash, react.__workletHash];
```

Comment:
this is invalid, consider following code:
```
if (dependencies === undefined) {
  const depInputs = Object.values(prepare._closure).concat(Object.values(react._closure));
  dependencies = [...depInputs, prepare.__workletHash, react.__workletHash];
} else {
  dependencies.push(prepare.__workletHash, react.__workletHash);
}
```
Pulling dependencies' values from ref results in using deprecated values(state's value won't update). Plus you should also use inputs from `react`, not only from `prepare`(`input` stored in ref pulls only from `prepare`)

Reply:
Yeah, make sense, will update code asap